app/src/main/java/io/intrepid/contest/screens/entrysubmission/entryname/EntryNameActivity.java

```diff
@@ -0,0 +1,75 @@
+package io.intrepid.contest.screens.entrysubmission.entryname;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.widget.Button;
+import android.widget.TextView;
+
+import butterknife.BindView;
+import butterknife.OnFocusChange;
+import butterknife.OnTextChanged;
+import io.intrepid.contest.R;
+import io.intrepid.contest.base.BaseMvpActivity;
+import io.intrepid.contest.base.PresenterConfiguration;
+
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.Presenter;
+import static io.intrepid.contest.screens.entrysubmission.entryname.EntryNameContract.View;
+
+public class EntryNameActivity extends BaseMvpActivity<Presenter> implements View {
```

Comment:
Consider calling this class JoinContestActivity or something more descriptive. It may also help avoiding long-ish view variable names (like contestantWelcomeTextView --> welcomeTextView)